validation/validator.go

```diff
@@ -129,19 +129,6 @@ func (v *Validator) castValue() mapstructure.DecodeHookFunc {
 			default:
 				castedValue, err = cast.ToSliceE(from.Interface())
 			}
-		case reflect.Map:
```

Comment:
The `map` structure is not fixed, so we can cast the value to `StringMapString`, `StringMapBool`, etc. And there is a BUG as well, `to.Type().Key().Kind()` checks the map key, but in the `case` logic, we are using `ToStringMapBoolE`, it should be `ToBoolMap***E`. 

I think we can remove the logic directly.